### ISSUE TITLE
Research PoS implementations

### ISSUE THREAD

**Issue by garious** [garious/loomination#3]
Is there any way we can incorporate an existing PoS implementation? Maybe Proof of Authority?

**Comment by garious** [comment]
No, we probably won't incorporate an existing PoS implementation. Thanks to proof of history, our proof of stake voting ends up being a fairly straightforward generalization of the *spending plan* contracts. We'll create a new entry type called Block that will point to an Entry ID and include a contract that does a 2/3 multisig vote. Once the votes are on the ledger, the entry and all prior may be coalesced. Note that the block the votes are in does not need to be finalized before the entries are coalesced. Signature verification is sufficient.

**Comment by garious** [comment]
Just to add to my previous comment, if the block that the votes are in needed to be finalized, then you'd never finalize. That also implies that votes need to survive rollback. That means that both leader and verifiers need to cache votes until they are in a block that's been finalized, and that implies that votes are broadcast to all nodes, not just the leader.

**Comment by garious** [comment]
Problems that need to solved here:

* Proof to anyone interpreting the chain that consensus has been reached
* Survive a leader handoff without leaving space for a timing attack (@aeyakovenko groks this better than I do)
* Calculate bond

So far my comments have only been about the first.

**Comment by garious** [comment]
Recommended reading from @ericlwilliams:
https://github.com/ethereum/wiki/wiki/Proof-of-Stake-FAQ
https://arxiv.org/abs/1406.5694

**Comment by bgorlick** [comment]
I was going to suggest their threshold relay as well as one option to consider. I'm giving it some thought still.

**Comment by garious** [comment]
Thanks @bgorlick, I'll be reviewing that closely. Excited to see this on Page 14:

```
For brevity, whenever in the protocol we say that a party uses the clock to update, τ, ep, and sl we mean
the following step:
Send (clock-read,sidC ) to Gclock; receive the current time τ and update ep := dτ /Re and slot index
sl = τ , accordingly.
```
